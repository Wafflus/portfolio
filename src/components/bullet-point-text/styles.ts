import styled from "styled-components";

export const BulletPointTextStyled = styled.li`
    color: ${({ theme }) => theme.data.text.lightColor};
    font-family: '${({ theme }) => theme.data.fontFamilies.outfit}', sans-serif;
    font-size: 1.125rem;
    letter-spacing: 0.1rem;
    line-height: 120%;
    list-style-type: none;
    position: relative;
    text-align: center;
    transition: transform 0.5s;

    &:hover {
        transform: scale(102%);
    }
    
    ${props => props.theme.breakpoint.above('lg',
        `
            &::before {
                background-color: ${props.theme.data.secondaryColor};
                border-radius: 50%;
                content: '';
                height: 20px;
                left: -40px;
                position: absolute;
                transition: background-color 0.5s;
                width: 20px;
            }

            &:hover::before {
                background-color: ${props.theme.data.primaryColor};
            }
        `
    )}
`;