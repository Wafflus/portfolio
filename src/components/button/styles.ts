import { IButtonStyledProps } from "./types";
import styled, { css } from "styled-components";

export const ButtonStyled = styled.a<IButtonStyledProps>`
    align-items: center;
    background-color: ${({ theme }) => theme.data.secondaryColor};
    border-radius: 8px;
    color: ${({ theme }) => theme.data.text.darkColor};
    display: flex;
    font-size: 2rem;
    font-variant: small-caps;
    gap: 16px;
    height: 66px;
    justify-content: center;
    letter-spacing: 0.1rem;
    max-width: 300px;
    min-width: 200px;
    padding: 8px;
    text-decoration: none;
    transition: transform 0.5s;
    width: 100%;

    ${props => props.outline && css`
        background-color: transparent;
        border: 1px solid ${({ theme }) => theme.data.primaryColor};
        color: #93855B;
    `}

    &:hover {
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
        transform: scale(102%);
    }
`;