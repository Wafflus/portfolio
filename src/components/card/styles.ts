import styled from "styled-components";

export const CardStyled = styled.article`
    border-radius: 8px;
    padding: 16px;

    &:nth-child(odd) {
        background-color: ${({ theme }) => theme.data.highlightColor};
    }

    &:hover {
        outline: 3px solid ${({ theme }) => theme.data.borderColor};

        & > div > img {
            border: 3px solid ${({ theme }) => theme.data.primaryColor};
            transform: scale(102%);
        }
    }

    ${props => props.theme.breakpoint.above('lg',
        `
            &:nth-child(even) > div {
                flex-direction: row-reverse;
            }
        `
    )}
`;

export const TitleStyled = styled.h3`
    color: ${({ theme }) => theme.data.text.darkColor};
    font-size: 1.5rem;
    font-variant: small-caps;
    font-weight: 600;
    letter-spacing: 0.1rem;
    text-align: center;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            font-size: 2rem;
            text-align: start;
        `
    )}
`;

export const ToolsStyled = styled.h4`
    color: ${({ theme }) => theme.data.text.defaultColor};
    font-variant: small-caps;
    letter-spacing: 0.1rem;
    text-align: center;
`;

export const DescriptionStyled = styled.p`
    color: ${({ theme }) => theme.data.text.defaultColor};
    flex: 1;
    letter-spacing: 0.1rem;
`;