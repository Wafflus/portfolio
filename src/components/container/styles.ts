import { IContainerStyledProps } from "./types";
import styled from "styled-components";

export const ContainerStyled = styled.div<IContainerStyledProps>`
    height: 100%;
    max-width: 1440px;
    padding: ${props => props.verticalPadding} ${props => props.horizontalPadding};
    width: 100%;
    z-index: 200;
`;