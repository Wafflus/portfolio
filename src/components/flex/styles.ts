import { IFlexStyledProps } from "./types";
import styled from "styled-components";

export const FlexStyled = styled.div<IFlexStyledProps>`
    align-items: ${props => props.defaultFlexProps?.alignItems};
    display: flex;
    flex-direction: ${props => props.defaultFlexProps?.flexDirection};
    gap: ${props => props.defaultFlexProps?.gap};
    justify-content: ${props => props.defaultFlexProps?.justifyContent};
    max-width: ${props => props.defaultFlexProps?.maxWidth};
    width: 100%;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            align-items: ${props.largeBreakpointFlexProps?.alignItems};
            flex-direction: ${props.largeBreakpointFlexProps?.flexDirection};
            gap: ${props.largeBreakpointFlexProps?.gap};
            justify-content: ${props.largeBreakpointFlexProps?.justifyContent};
            max-width: ${props.largeBreakpointFlexProps?.maxWidth};
        `
    )}
`;