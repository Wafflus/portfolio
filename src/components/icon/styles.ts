import styled from "styled-components";

import { IImageProps } from "./types";

export const IconStyled = styled.img<IImageProps>`
    width: ${props => props.size}${props => props.unit};
    height: ${props => props.size}${props => props.unit};
    border-radius: 50%;
`;