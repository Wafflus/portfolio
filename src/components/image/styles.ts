import styled from "styled-components";

export const ImageStyled = styled.img`
    border-radius: 8px;
    border: 3px solid transparent;
    height: 100%;
    max-height: 350px;
    max-width: 500px;
    transition: transform 0.5s, border 0.5s;
    width: 100%;
`;