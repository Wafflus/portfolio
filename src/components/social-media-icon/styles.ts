import { ISocialMediaIconTextStyledProps } from "./types";
import styled, { css } from "styled-components";

export const AnchorStyled = styled.a`
    border-radius: 50%;
    cursor: pointer;
    height: 64px;
    outline: 0px solid ${({ theme }) => theme.data.borderColor};
    text-decoration: none;
    transition: transform 1s, outline 0.5s;
    width: 64px;

    &:hover {
        outline: 10px solid ${({ theme }) => theme.data.borderColor};
        position: relative;
        transform: translateY(-16px);
    }
`;

export const TextStyled = styled.span<ISocialMediaIconTextStyledProps>`
    color: ${({ theme }) => theme.data.text.darkColor};
    font-family: '${({ theme }) => theme.data.fontFamilies.outfit}', sans-serif;
    letter-spacing: 10%;
    opacity: 0;
    text-align: center;
    transition: opacity 0.5s;

    ${props => props.active && css`
        opacity: 1;
    `}
`;

export const DivStyled = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
`;