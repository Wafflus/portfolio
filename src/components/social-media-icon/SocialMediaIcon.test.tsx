import "jest-styled-components";
import { fireEvent, render } from "@testing-library/react";
import { socials } from "data/social-data";
import { theme } from "styles";
import { ThemeProvider } from "styled-components";
import { SocialMediaIcon } from ".";

test('social media icon renders', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <SocialMediaIcon alt="Alternative text." src={socials.github} href={'https://github.com/Wafflus'}>
                <>Github</>
            </SocialMediaIcon>
        </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toBeInTheDocument();
});

test('text should not show up', () => {
    const socialMediaIcon = render(
        <ThemeProvider theme={theme}>
            <SocialMediaIcon alt="Alternative text." src={socials.github} href={'https://github.com/Wafflus'}>
                <>Github</>
            </SocialMediaIcon>
        </ThemeProvider>
    );

    const text = socialMediaIcon.getByText('Github').closest('span');

    expect(text).toHaveStyleRule('opacity', '0');
});

test('text should show when anchor is hovered', () => {
    const socialMediaIcon = render(
        <ThemeProvider theme={theme}>
            <SocialMediaIcon alt="Alternative text." src={socials.github} href={'https://github.com/Wafflus'}>
                <>Github</>
            </SocialMediaIcon>
        </ThemeProvider>
    );

    const anchor = socialMediaIcon.getByRole('link');
    const text = socialMediaIcon.getByText('Github').closest('span');

    fireEvent.mouseEnter(anchor);

    expect(text).toHaveStyleRule('opacity', '1');
});

test('anchor should have different styles when hovered', () => {
    const socialMediaIcon = render(
        <ThemeProvider theme={theme}>
            <SocialMediaIcon alt="Alternative text." src={socials.github} href={'https://github.com/Wafflus'}>
                <>Github</>
            </SocialMediaIcon>
        </ThemeProvider>
    );

    const anchor = socialMediaIcon.getByRole('link').closest('a');
    const options = { modifier: ':hover' };

    expect(anchor).toHaveStyleRule('outline', `10px solid ${theme.data.borderColor}`, options);
    expect(anchor).toHaveStyleRule('position', 'relative', options);
    expect(anchor).toHaveStyleRule('transform', 'translateY(-16px)', options);
});