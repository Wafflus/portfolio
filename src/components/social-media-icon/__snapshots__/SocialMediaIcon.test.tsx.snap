// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`social media icon renders 1`] = `
.c1 {
  border-radius: 50%;
  cursor: pointer;
  height: 64px;
  outline: 0px solid #FFF0BF;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: -webkit-transform 1s,outline 0.5s;
  -webkit-transition: transform 1s,outline 0.5s;
  transition: transform 1s,outline 0.5s;
  width: 64px;
}

.c1:hover {
  outline: 10px solid #FFF0BF;
  position: relative;
  -webkit-transform: translateY(-16px);
  -ms-transform: translateY(-16px);
  transform: translateY(-16px);
}

.c3 {
  color: #756A47;
  font-family: 'Outfit',sans-serif;
  -webkit-letter-spacing: 10%;
  -moz-letter-spacing: 10%;
  -ms-letter-spacing: 10%;
  letter-spacing: 10%;
  opacity: 0;
  text-align: center;
  -webkit-transition: opacity 0.5s;
  transition: opacity 0.5s;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  width: 64px;
  height: 64px;
  border-radius: 50%;
}

<div
  class="c0"
>
  <a
    class="c1"
    href="https://github.com/Wafflus"
    target="_blank"
  >
    <img
      alt="Alternative text."
      class="c2"
      size="64"
      src="https://github.com/Wafflus"
    />
  </a>
  <span
    class="c3"
  >
    Github
  </span>
</div>
`;
